pkg/types/powervs/machinepools.go

```diff
@@ -0,0 +1,51 @@
+package powervs
+
+// MachinePool stores the configuration for a machine pool installed on IBM Power VS.
+type MachinePool struct {
+	// VolumeIDs is the list of volumes attached to the instance.
+	//
+	// +optional
+	VolumeIDs []string `json:"volumeIDs"`
+
+	// Memory defines the memory in GB for the instance.
+	//
+	// +optional
+	Memory string `json:"memory"`
+
+	// Processors defines the processing units for the instance.
+	//
+	// +optional
+	Processors string `json:"processors"`
+
+	// ProcType defines the processor sharing model for the instance.
+	//
+	// +optional
+	ProcType string `json:"procType"`
```

Comment:
Use a type with known values. Detail the accepted values in the description of the field. Add kubebuilder validation on the type.